### PR TITLE
Labels `VolumeSnapshotClass`

### PR DESCRIPTION
**Pull request by raaizik**
## Changes

- Labels `VolumeSnapshotClass` with RDR required parameters
[RHSTOR-5795](https://issues.redhat.com/browse/RHSTOR-5795)

### CODE REVIEW

File: internal/controller/storageclaim_controller.go [hunk 1]
```diff
@@ -408,14 +409,14 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
...
+				volumeSnapshotClass = r.getCephDriverVolumeSnapshotClass(resource.Labels, resource.Annotations)
 				utils.AddAnnotation(volumeSnapshotClass, storageClaimAnnotation, r.storageClaim.Name)
-				if err := r.createOrReplaceVolumeSnapshotClass(volumeSnapshotClass); err != nil {
-					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
+				err = utils.CreateOrReplace(r.ctx, r.Client, volumeSnapshotClass, func() error {
+					volumeSnapshotClass.Parameters = data
+					return nil
+				})
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
```
Comment (Madhu-1) [inline internal/controller/storageclaim_controller.go:419]: it should be volumeSnapshotClass instead of stroageclass here

**Review by Madhu-1 — COMMENTED** [review]
LGTM, small nit

File: internal/controller/storageclaim_controller.go [hunk 2]
```diff
@@ -523,68 +522,20 @@ func (r *StorageClaimReconciler) getCephRBDStorageClass(data map[string]string)
...
 
-func (r *StorageClaimReconciler) getCephFSVolumeSnapshotClass(data map[string]string) *snapapi.VolumeSnapshotClass {
+func (r *StorageClaimReconciler) getCephDriverVolumeSnapshotClass(
+	labels map[string]string, annotations map[string]string) *snapapi.VolumeSnapshotClass {
 	volumesnapshotclass := &snapapi.VolumeSnapshotClass{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: r.storageClaim.Name,
+			Name:        r.storageClaim.Name,
+			Labels:      labels,
+			Annotations: annotations,
```
Comment (Madhu-1) [inline internal/controller/storageclaim_controller.go:531]: getCephDriverVolumeSnapshotClass is only supported for cephfs not for RBD? the Driver name is hardcoded to cephfs driver.

Comment (raaizik) [reply internal/controller/storageclaim_controller.go:537]: Yeah, I left out some of the code now it's back

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM, please let me know once the testing is done

**Comment by rewantsoni** [comment]
Testing results:
```
Before updating the operator image
[rewantsoni scripts] k get sc ocs-storagecluster-ceph-rbd -oyaml | yq '.metadata.labels' && k get sc ocs-storagecluster-cephfs -oyaml | yq '.metadata.labels' && k get volumesnapshotclasses.snapshot.storage.k8s.io ocs-storagecluster-ceph-rbd -oyaml | yq '.metadata.labels' && k get volumesnapshotclasses.snapshot.storage.k8s.io ocs-storagecluster-cephfs -oyaml | yq '.metadata.labels'
null
null
null
null

After updating the operator image
[rewantsoni scripts] k get sc ocs-storagecluster-ceph-rbd -oyaml | yq '.metadata.labels' && k get sc ocs-storagecluster-cephfs -oyaml | yq '.metadata.labels' && k get volumesnapshotclasses.snapshot.storage.k8s.io ocs-storagecluster-ceph-rbd -oyaml | yq '.metadata.labels' && k get volumesnapshotclasses.snapshot.storage.k8s.io ocs-storagecluster-cephfs -oyaml | yq '.metadata.labels'
ramendr.openshift.io/storageID: 4316f047b30670e91f804f37a7faceb7
ramendr.openshift.io/storageID: bb1a42e6f5de8381f971418cfe1ec72f
ramendr.openshift.io/storageID: 4316f047b30670e91f804f37a7faceb7
ramendr.openshift.io/storageID: bb1a42e6f5de8381f971418cfe1ec72f

After deleting the volumesnapshot object
[rewantsoni scripts] k delete volumesnapshotclasses.snapshot.storage.k8s.io ocs-storagecluster-ceph-rbd
volumesnapshotclass.snapshot.storage.k8s.io "ocs-storagecluster-ceph-rbd" deleted

Waitied a few mins for reconcile and 
[rewantsoni scripts] k get volumesnapshotclasses.snapshot.storage.k8s.io                                   
NAME                          DRIVER                                  DELETIONPOLICY   AGE
csi-aws-vsc                   ebs.csi.aws.com                         Delete           8h
ocs-storagecluster-ceph-rbd   openshift-storage.rbd.csi.ceph.com      Delete           36s
ocs-storagecluster-cephfs     openshift-storage.cephfs.csi.ceph.com   Delete           5m15s
```

**Pull request merged**